### DISCUSSION TITLE
add option to delete the selected element

### DISCUSSION THREAD

**Discussion by AmosSzust** [bcakmakoglu/vue-flow#1230] [Ideas]
Hi

Like we have `this.vueFlowInstance.getSelectedElements`, please add `this.vueFlowInstance.deleteSelectedElements`
Thanks for a wonderful component.

**Comment by bcakmakoglu** [comment]
You'll be shocked to hear that [`removeSelectedElements`](https://vueflow.dev/typedocs/interfaces/Actions.html#removeselectedelements) exists.

**Reply by AmosSzust** [reply]
I'm not shocked, I know `removeSelectedElements`, which unselect selected elements, exists.
I'm shocked that `this.vueFlowInstance.deleteSelectedElements` that should **delete** selected element(s), doesn't exist :)

**Reply by bcakmakoglu** [reply]
You weren't really specific in your opening post.

You can just do `removeNodes(getSelectedNodes.value)`.
Connected edges of a node are removed as well using `removeNodes`

```ts 
type RemoveNodes = (
  nodes: (string | Node) | (Node | string)[] | ((nodes: GraphNode[]) => (string | Node) | (Node | string)[]),
  removeConnectedEdges?: boolean,
  removeChildren?: boolean,
) => void
```

An extra wrapper just do to `removeNodes(getSelectedNodes.value)` for you seems highly unnecessary to me.

**Reply by AmosSzust** [reply]
Thanks for this. 2 things:
1. The advantage of selectedElements is that it allows you to delete either a node or a connection line, if I understand correctly, removeNodes only removes selected nodes, what if the user selected a connection line?
2. I have the same issue as I reported here: https://github.com/bcakmakoglu/vue-flow/discussions/1229 , see screenshot. If I view `this.vueFlowInstance.getSelectedNodes`, I see the selected node, if I view `this.vueFlowInstance.getSelectedNodes.value` it gives me undefined and obviously fails.

![image](https://github.com/bcakmakoglu/vue-flow/assets/100950035/e9459f86-b951-4036-acf9-41aa84d4fc51)

Thank you for a magnificent component!

**Reply by bcakmakoglu** [reply]
You'd be better off handling both element types separately instead of using a single handler for both types (nodes/edges).
For one, the concept of a unified elements array (which doesn't even exist in the state under the hood) is gonna be dropped in the next major anyway, so there won't be an introduction of a helper function like `removeElements` and I'll start deprecating these element related APIs soon 😅

So in your case sth like this

```ts
function deleteSelectedElements() {
	const selectedNodes = getSelectedNodes.value;

	const selectedEdges = getSelectedEdges.value;

	if (selectedNodes.length) {
		removeNodes(selectedNodes)
	}

	if (selectedEdges.length) {
		removeEdges(selectedEdges)
	}
}
```

**Reply by AmosSzust** [reply]
Thanks for the tip, what about the issue from the screenshot?
If I do `useVueFlow().getSelectedNodes.value`, this is what I get and the result is an empty array while there is one selected node.
![image](https://github.com/bcakmakoglu/vue-flow/assets/100950035/7a94a4c2-1c9a-47bf-b555-774170c9516b)
